src/cmstats.c

```diff
@@ -49,6 +50,26 @@ s_duplicator (const void *self)
     return (void*) fty_proto_dup ((fty_proto_t*) self);
 }
 
+static double
+safe_atof (const char *str)
+{
+    char *endptr = (char *) zmalloc (sizeof (char));
+    errno = 0;
+    double ret = strtod (str, &endptr);
```

Comment:
This looks fishy: according to common logic of such functions (and UPDATE: docs at https://www.tutorialspoint.com/c_standard_library/c_function_strtod.htm too) the `endptr` should just be an (uninitialized) `char *` where the `strtod()` returns a NULL or pointer to the non-numeric character from the tail of `str`. I guess if all `str` was processed into a number, the `endptr` will point to the `\0` of `str`.

In short, no `malloc()` and no `free()` are needed.